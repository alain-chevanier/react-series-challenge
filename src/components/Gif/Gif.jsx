import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import favoriteGifsActions from '../../actions/favoriteGifs';

const Gif = ({ object, addFavorite, removeFavorite }) => (
  <div>
    <h3>{object.title}</h3>
    <iframe src={object.embed_url} title={object.title} width="100%" height="100%" />
    <button onClick={() => addFavorite({ payload: { gif: object } })}>
      Add to favorite
    </button>
    <button onClick={() => removeFavorite({ payload: { gif: object } })}>
      Remove from favorite
    </button>
  </div>
);

Gif.propTypes = {
  object: PropTypes.shape({
    embed_url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
  }).isRequired,
  addFavorite: PropTypes.func.isRequired,
  removeFavorite: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => {
  const { addFavorite, removeFavorite } = favoriteGifsActions.creators;
  return bindActionCreators({
    addFavorite,
    removeFavorite,
  }, dispatch);
};

export default connect(null, mapDispatchToProps)(Gif);

