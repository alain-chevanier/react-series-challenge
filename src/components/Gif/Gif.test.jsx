import React from 'react';
import { shallow } from 'enzyme';
import Gif from './Gif';

describe('Gif component', () => {
  it('renders the liked image', () => {
    const obj = {
      title: 'hello',
      embed_url: 'url',
      id: 'XX',
      isFavorite: true,
    };
    const mockAdd = jest.fn();
    const mockRemove = jest.fn();
    const wrapper = shallow(<Gif object={obj} addFavorite={mockAdd} removeFavorite={mockRemove} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('renders the liked image', () => {
    const obj = {
      title: 'hello',
      embed_url: 'url',
      id: 'XX',
      isFavorite: false,
    };
    const mockAdd = jest.fn();
    const mockRemove = jest.fn();
    const wrapper = shallow(<Gif object={obj} addFavorite={mockAdd} removeFavorite={mockRemove} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
