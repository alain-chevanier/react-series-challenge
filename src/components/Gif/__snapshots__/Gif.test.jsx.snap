// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif component renders its children within the containing div 1`] = `
"<styled.div>
  <styled.p>
    hello
  </styled.p>
  <styled.iframe src=\\"url\\" title=\\"hello\\" />
  <styled.img src=\\"favorite-heart-like-selected.png\\" alt=\\"Remove from favorite\\" onClick={[Function: onClick]} />
</styled.div>"
`;

exports[`Gif component renders the liked image 1`] = `
"<styled.div>
  <styled.p>
    hello
  </styled.p>
  <styled.iframe src=\\"url\\" title=\\"hello\\" />
  <styled.img src=\\"favorite-heart-like-selected.png\\" alt=\\"Remove from favorite\\" onClick={[Function: onClick]} />
</styled.div>"
`;

exports[`Gif component renders the liked image 2`] = `
"<styled.div>
  <styled.p>
    hello
  </styled.p>
  <styled.iframe src=\\"url\\" title=\\"hello\\" />
  <styled.img src=\\"favorite-heart-like-container.png\\" alt=\\"Add to favorite\\" onClick={[Function: onClick]} />
</styled.div>"
`;

exports[`Gif component renders the liked image 3`] = `
"<styled.div>
  <styled.p>
    hello
  </styled.p>
  <styled.iframe src=\\"url\\" title=\\"hello\\" />
  <styled.img src=\\"favorite-heart-like-container.png\\" alt=\\"Add to favorite\\" onClick={[Function: onClick]} />
</styled.div>"
`;
